File: arreglo.py
class Arreglo:
    def __init__(self) :
        self.aux=[]
        self.valores=[]
        self.tamaño=0;
        self.suma=0;
        self.Empezar()

    def Empezar(self):
        self.tamaño= (int) (input("ingrese el tamaño que desea "))   
        for i in range(self.tamaño):
            self.valores.append((int) (input()))

        for i in range(self.tamaño-1):
            for j in range((i+1),self.tamaño):
                self.suma=self.Suma(i,j)
                if self.suma ==0:
                    self.aux.append((j-i)+1)
                    break; 

        self.MetodoQuicSort()
        print(self.aux[0])            

    def MetodoQuicSort(self):
        pivote=0
        for i in range(1,len(self.aux)):
            j=i
            while j>0 and self.aux[j-1]>self.aux[j]:
                pivote =self.a[j-1]
                self.aux[j-1]=self.aux[j]
                self.aux[j]= pivote
                j-=1

    def Suma(self,i,j): 
        suma=0
        for x in range(i,(j+1)):
            suma += self.valores[x] 
        return suma



if __name__ == "__main__":
    Arreglo()

File: Arreglo.java
import java.util.*;

public class Arreglo {
    //Arraylist<int> arrayAux = new Arraylist<int>();
    static ArrayList<Integer> Aux = new ArrayList<Integer>();
    public static void main(String[] args) {
        
        int pos = 0;
        int n = 0;
        int suma = 0;
        int aux[] = new int[10];

        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();

        int a[] = new int[n];

        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        Aux.add(2);
        /*
         * for(int i=0; i<n;i++){ int j=i+1; while(j<n-1){ suma = sum(a,i,j); if (suma
         * ==0){ System.out.println("entra porque sum es 0 "+" j e i "+(j-i)); aux[pos]
         * = (j-i); pos++; break; }else{ j++; } } }
         * 
         * for(int i=0;i<aux.length;i++){ System.out.println(aux[i]); }
         */

        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {

                suma = sum(a, i, j);

                if (suma == 0) {
                    aux[pos] = j - i;
                    pos++;
                    break;
                }
            }
        }
        MetodoInsertSort(aux);
        System.out.println(aux[0]);
    }

    public static void MetodoInsertSort(int array[]) {
        int pibote;
        for (int i = 1; i < (array.length); i++) {
            int j = i;
            while (j > 0 && array[j - 1] > array[j]) {
                pibote = array[j - 1];
                array[j - 1] = array[j];
                array[j] = pibote;
                j--;
            }
        }
    }

    public static int sum(int a[], int i, int j) {

        int suma = 0;

        for (int x = i; x <= j; x++) {

            suma = suma + a[x];

        }

        return suma;
    }
    /*
     * public static void main(String[] args) { Subsecuencia sub = new Subsecuencia
     * ();
     * 
     * 
     * 
     * } /*int arregloNum[]; Scanner scanNum = new Scanner(System.in);
     * 
     * System.out.println("De que tamaño es el arreglo?"); int tamArray =
     * scanNum.nextInt(); arregloNum=new int [tamArray];
     * 
     * System.out.println("Digite los numeros del arreglo");
     * 
     * for(int i=0 ; i<tamArray ; i++){
     * 
     * }
     */

}